_rules/SC4-1-2-aria-state-or-property-has-valid-value.md

```diff
@@ -0,0 +1,270 @@
+---
+name: ARIA state or property has valid value
+test_type: atomic
+
+description: |
+   This rule checks that each ARIA state or property has a valid value
+   
+success_criterion:
+- 4.1.2 # Name, Role, Value
+
+test_aspects:
+- DOM Tree
+- CSS Styling
+
+authors:
+- Wilco Fiers
+- Anne Thyme Nørregaard
+---
+
+## Test procedure
+
+### Applicability
+
+Any [non-empty](#non-empty) [WAI-ARIA 1.1 state or property](https://www.w3.org/TR/wai-aria-1.1/#state_prop_def) that is specified on an HTML or SVG element.
+
+### Expectation
+
+Each test target has a valid value according to its [WAI-ARIA 1.1 value type](https://www.w3.org/TR/wai-aria-1.1/#propcharacteristic_value).
+
+For value types `ID Reference` and `ID Reference List`, a valid value for a [WAI-ARIA required property](https://www.w3.org/TR/wai-aria-1.1/#requiredState) requires that at least one of the elements with the given ids exists in the same [document tree](https://www.w3.org/TR/dom41/#document-trees) or [shadow tree](https://www.w3.org/TR/dom41/#shadow-trees) as the element that specifies the target attribute. For non-required properties, this is not a requirement.
```

Comment:
The first sentence reads like a statement rather than an expectation. It is also a long and confusing sentence. Maybe say:

> For value types `ID Reference` and `ID Reference List`, at least one of the elements with the given ids exists in the same [document tree](https://www.w3.org/TR/dom41/#document-trees) or [shadow tree](https://www.w3.org/TR/dom41/#shadow-trees) as the element that specifies the target attribute.

The part "a valid value for a [WAI-ARIA required property](https://www.w3.org/TR/wai-aria-1.1/#requiredState) requires that" was removed. If this is needed, then it should be a note like this:

> Note: This behavior is defined in [WAI-ARIA required property](https://www.w3.org/TR/wai-aria-1.1/#requiredState)